File: server/src/main/java/com/fluffy/auth/domain/OAuth2Provider.java
package com.fluffy.auth.domain;

import com.fluffy.global.exception.NotFoundException;
import java.util.Arrays;

public enum OAuth2Provider {

    GITHUB,
    GOOGLE,
    ;

    public static OAuth2Provider from(String provider) {
        return Arrays.stream(values())
                .filter(it -> it.name().equalsIgnoreCase(provider))
                .findFirst()
                .orElseThrow(() -> new NotFoundException("존재하지 않는 OAuth2 제공자입니다. 제공자: " + provider));
    }
}

File: server/src/main/kotlin/com/fluffy/auth/domain/OAuth2Provider.kt
package com.fluffy.auth.domain

import com.fluffy.global.exception.NotFoundException

enum class OAuth2Provider {

    GITHUB,
    GOOGLE,
    ;

    companion object {

        fun from(provider: String): OAuth2Provider {
            return entries.find { it.name.equals(provider, ignoreCase = true) }
                ?: throw NotFoundException("존재하지 않는 OAuth2 제공자입니다. 제공자: $provider")
        }
    }
}